lib/gyp/generator/ninja/index.js

```diff
@@ -395,7 +396,12 @@ Ninja.prototype.actions = function actions() {
 
   let res = [];
   list.forEach((action) => {
-    const actionRule = action.action_name + '_' + this.index;
+    const safeActionName = action.action_name.replace(/\s/g, '_');
+    global.action_names = (global.action_names || {});
```

Comment:
It should store `action_names` in some JS instance, perhaps `NinjaMain`. Using `global` is a no-go here. Sorry!